src/legacy/plugin_discovery/plugin_spec/plugin_spec_options.d.ts

```diff
@@ -21,16 +21,19 @@ import { Capabilities } from '../../../core/public';
 
 export type InitPluginFunction = (server: Server) => void;
 export interface UiExports {
-  injectDefaultVars: (server: Server) => { [key: string]: any };
+  injectDefaultVars?: (server: Server) => { [key: string]: any };
   styleSheetPaths?: string;
+  visTypes?: string[];
+  interpreter?: string[];
+  hacks?: string[];
 }
 
 export interface PluginSpecOptions {
   id: string;
-  require: string[];
-  publicDir: string;
+  require?: string[];
+  publicDir?: string;
   uiExports?: UiExports;
   uiCapabilities?: Capabilities;
-  init: InitPluginFunction;
-  config: any;
+  init?: InitPluginFunction;
+  config?: any;
```

Comment:
These are all definitely optional?

Just asking -- I don't know the answer 😛

Reply:
they seem to be, we don't pass them in with a few plugins